test/util/metrics.ts

```diff
@@ -0,0 +1,259 @@
+import DataOperation from '../../app/models/data-operation';
+import db from '../../app/util/db';
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { parseSchemaFile } from '../helpers/data-operation';
+import { buildJob } from '../helpers/jobs';
+import { getProductMetric, getRequestMetric, getResponseMetric } from '../../app/util/metrics';
+import { JobStatus } from '../../app/models/job';
+
+const operation = new DataOperation(parseSchemaFile('valid-operation-input.json'));
+
+describe('Metrics construction', function () {
+  describe('Getting the request metric', function () {
+    describe('when an operation contains all fields', function () {
+      const metric = getRequestMetric(operation, 'a neat service');
+
+      it('includes all of the fields in the metric', function () {
+        expect(Object.keys(metric)).to.eql([
+          'request_id', 'user_id', 'parameters', 'bbox', 'rangeBeginDateTime', 'rangeEndDateTime',
+        ]);
+      });
+
+      it('sets the request ID correctly', function () {
+        expect(metric.request_id).to.equal('c045c793-19f1-43b5-9547-c87a5c7dfadb');
```

Comment:
I think it would be better to reference the values (like 'c045c793-19f1-43b5-9547-c87a5c7dfadb') from the fields on `operation` here and elsewhere instead of hard-coding them.